### PR TITLE
Make OVMS adhere to OV naming convention

### PR DESCRIPTION
**Pull request by atobiszei**
Blob->Tensor
ExecutableNetwork -> CompiledModel
CNNetwork -> Model

JIRA:CVS-75095

### CODE REVIEW

File: src/deserialization.hpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ class ConcreteTensorProtoDeserializator {
             }
             ov::runtime::Tensor tensor(ov::element::f16, shape);
             // Needs conversion due to zero padding for each value:
-            // https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/core/framework/tensor.proto#L55
+            // https://github.com/tensorflow/tensorflow/tensor/v2.2.0/tensorflow/core/framework/tensor.proto#L55
```
Comment (dkalinowski) [inline src/deserialization.hpp:60]: Links are not working

File: src/modelinstance.hpp [hunk 2]
```diff
@@ -285,7 +285,7 @@ class ModelInstance {
     Status recoverFromReloadingError(const Status& status);
 
     /**
-         * @brief Perform full engine/network reload with dynamic parameter
+         * @brief Perform full engine/model reload with dynamic parameter
```
Comment (dkalinowski) [inline src/modelinstance.hpp:288]: engine/model is kinda outdated, engine refered to ieCore which is singleton now

File: src/predict_request_validation_utils.cpp [hunk 3]
```diff
@@ -204,7 +204,7 @@ Status RequestValidator::validateTensorContentSize(const tensorflow::TensorProto
     double      data in request.tensor_content
 
     _TENSOR_CONTENT_TYPES
-    https://github.com/tensorflow/tensorflow/blob/903a6399aab19b549fefd0ead836af644f3d00f8/tensorflow/python/framework/tensor_util.py#L237
+    https://github.com/tensorflow/tensorflow/tensor/903a6399aab19b549fefd0ead836af644f3d00f8/tensorflow/python/framework/tensor_util.py#L237
```
Comment (dkalinowski) [inline src/predict_request_validation_utils.cpp:207]: link not working here as well

File: src/status.hpp [hunk 4]
```diff
@@ -50,19 +50,19 @@ enum class StatusCode {
...
+    RESHAPE_REQUIRED,                              /*!< Model instance needs to be reloaded with new shape */
+    BATCHSIZE_CHANGE_REQUIRED,                     /*!< Model instance needs to be reloaded with new batch size */
+    FORBIDDEN_MODEL_DYNAMIC_PARAMETER,             /*!< Value of the provided param is forbidden */
+    ANONYMOUS_FIXED_SHAPE_NOT_ALLOWED,             /*!< Anonymous fixed shape is invalid for models with multiple inputs */
+    ANONYMOUS_FIXED_LAYOUT_NOT_ALLOWED,            /*!< Anonymous fixed layout is invalid for models with multiple inputs */
+    CONFIG_SHAPE_IS_NOT_IN_NETWORK,                /*!< Configured tensor shape is not present in network */
+    CONFIG_LAYOUT_IS_NOT_IN_NETWORK,               /*!< Configured tensor layout is not present in network */
+    CONFIG_SHAPE_MAPPED_BUT_USED_REAL_NAME,        /*!< Using old name of input/output in config shape when mapped in mapping_config.json*/
+    CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME,       /*!< Using old name of input/output in config layout when mapped in mapping_config.json*/
+    CANNOT_LOAD_COMPILED_MODEL_INTO_TARGET_DEVICE,
```
Comment (dkalinowski) [inline src/status.hpp:65]: CANNOT_LOAD_MODEL_INTO_TARGET_DEVICE would fit better here. Model becomes CompiledModel only if the loading into device succeeds.

Comment (dkalinowski) [reply src/status.hpp:65]: We do not load CompiledModel into target device. We load Model into target device.

File: src/predict_request_validation_utils.cpp (hunk 3 above)
Comment (atobiszei) [inline src/predict_request_validation_utils.cpp:207]: fix

File: src/test/azurefilesystem_test.cpp [hunk 5]
```diff
@@ -51,17 +51,17 @@ std::string getAnonFilePath() { return std::string("azfs://my-sample-share/my-sa
 
 std::string getAnonDirPath() { return std::string("azfs://my-sample-share/my-sample-directory"); }
 
-std::string getAnonBlobFilePath() { return std::string("az://my-sample-container/my-sample-directory/my_test_file.txt"); }
+std::string getAnonTensorFilePath() { return std::string("az://my-sample-container/my-sample-directory/my_test_file.txt"); }
```
Comment (atobiszei) [inline src/test/azurefilesystem_test.cpp:54]: WHole file changes to be reverted

File: src/test/get_model_metadata_response_test.cpp [hunk 6]
```diff
@@ -59,7 +59,7 @@ class GetModelMetadataResponse : public ::testing::Test {
     tensor_desc_map_t inputTensors;
     tensor_desc_map_t outputTensors;
     ovms::tensor_map_t networkInputs;
```
Comment (dkalinowski) [inline src/test/get_model_metadata_response_test.cpp:61]: Why networkInputs but servableOutputs?

File: src/status.hpp (hunk 4 above)
Comment (atobiszei) [reply src/status.hpp:65]: we load ov::CompiledModel into target device:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L614

Comment (dkalinowski) [reply src/status.hpp:65]: We get CompiledModel as a result of loading Model into target device:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L592

Comment (atobiszei) [reply src/status.hpp:65]: We first create model from files for OV:
        ``model = loadOVModelPtr(modelFile);``
        https://github.com/openvinotoolkit/model_server/blob/69e2b070c456099fc4486978962faa03adf7d0d5/src/modelinstance.cpp#L531
Here there is no target device.

Then we get compiled model by compiling it onto target device in:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L592

See also where this error is returned:
https://github.com/openvinotoolkit/model_server/blob/69e2b070c456099fc4486978962faa03adf7d0d5/src/modelinstance.cpp#L616

It is returned when we failed to load compile model.

Maybe we should alter the naming of those functions as well to:

Read model -> load model files // which does only load the layers and weights into RAM but not on inference device
Compile(?Load) model -> compile model onto target device // which compiles model & loads into inference device

Comment (dkalinowski) [reply src/status.hpp:65]: Everything you have stated above is true, but summarizes to the current wrong status code name:
```
CANNOT_LOAD_COMPILED_MODEL_INTO_TARGET_DEVICE
```
Cannot load compiled model into target device - means we HAVE compiled model and we pass it to device to load it. This is not true, we have MODEL and pass it to device and take COMPILED MODEL in return.

You can also see that in API 1.0 we had:
```
CANNOT_LOAD_NETWORK_INTO_TARGET_DEVICE
```
but not `CANNOT_LOAD_EXECUTABLE_NETWORK_INTO_TARGET_DEVICE` which would follow your naming convention used right now in this MR.

This is really minor, but I suggest changing it to:
```
CANNOT_LOAD_MODEL_INTO_TARGET_DEVICE
```
and the status message as well to match what it exactly does under the hood

**Review by dkalinowski — APPROVED** [review]

File: src/test/stateful_test_utils.hpp [hunk 7]
```diff
@@ -73,18 +73,18 @@ class DummyStatefulModel {
...
 
     std::shared_ptr<ov::Model> network;
-    std::shared_ptr<ov::runtime::CompiledModel> execNetwork;
+    std::shared_ptr<ov::runtime::CompiledModel> compiledModel;
 
     const std::string stateName = "state";
 
 public:
     DummyStatefulModel() {
         network = ieCore.read_model(MODEL_PATH);
```
Comment (mzegla) [inline src/test/stateful_test_utils.hpp:82]: Here and in multiple other places `ieCore::read_model` output is saved in a variable called `network`.
Is that okay or should these variables be renamed to `model`?

**Review by mzegla — APPROVED** [review]

**Pull request merged**